src/index.ts

```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
+const octokit = require('@octokit/rest')();
+const createGithubService = require('./createGithubService').default;
+const deploy = require('./deploy').default;
+const exec = require('child_process').exec;
+
+async function sgAutorelease({rootPath, sourceDirectory, domain, repoOwner, repoName, githubToken, pr}) {
+  const githubServiceParams = {
+    owner: repoOwner,
+    repo: repoName,
+    token: githubToken
+  };
+  const valueExists = (value): boolean => !!value;
+  const githubServiceParamsAreValid = Object.values(githubServiceParams).every(valueExists);
+  const githubService = githubServiceParamsAreValid && createGithubService(octokit, githubServiceParams);
```

Comment:
see the comment in `deploy.ts` - I think it would be more readable to assume the `githubService` always exists. maybe adding some `isInitialized()` on `githubService` will be a more declarative way to check it